src/components/Form/ProductCreateAndEditForm.vue

```diff
@@ -171,11 +178,16 @@ const productsTableValues = (productsContentValue) => {
 };
 const isFormValid = async () => {
   const { valid } = await form.value.validate();
-  return (
-    props.productInfo.resourcesContent?.length > 0 &&
-    props.productInfo.productionNumber?.length > 0 &&
-    valid
-  );
+  return valid;
+};
+
+const isResourceSelected = () => {
+  const isResourceSelected = !!props.productInfo.resourcesContent?.length;
+  if (!isResourceSelected || isResourceSelected <= 0) {
+    snackbarProvider.showErrorSnackbar("Please select at least 1 resource!");
+    return false;
+  }
+  return true;
 };
```

Comment:
This function has redundant checks. The condition !!props.productInfo.resourcesContent?.length will already return false if the length is 0 or the property is undefined.
We can simplify it like this:
`const isResourceSelected = () => {
  if (!props.productInfo.resourcesContent?.length) {
    snackbarProvider.showErrorSnackbar("Please select at least 1 resource!");
    return false;
  }
  return true;
};
`